### ISSUE TITLE
Package install fails on macOS 12.3b

### ISSUE THREAD

**Issue by MagerValp** [freegeek-pdx/mkuser#2]
When an mkuser created packge is installed on macOS 12.3 beta, the `preinstall` script fails:

```
mkuser PREINSTALL PACKAGE: Starting...
mkuser PREINSTALL PACKAGE: Checking if user can be created before doing anything...
mkuser PREINSTALL PACKAGE: Creating extracted resources directory...
mkuser PREINSTALL PACKAGE: Extracting user picture...
mkuser PREINSTALL PACKAGE: Extracting passwords deobfuscation script...
bad decrypt
4374381952:error:06FFF064:digital envelope routines:CRYPTO_internal:bad decrypt:/System/Volumes/Data/SWE/macOS/BuildRoots/26a668749f/Library/Caches/com.apple.xbs/Sources/libressl/libressl-76.100.6/libressl-3.3/crypto/evp/evp_enc.c:549:
zcat: unknown compression format
mkuser PREINSTALL PACKAGE ERROR: Failed to decrypt passwords deobfuscation script (THIS SHOULD NOT HAVE HAPPENED, PLEASE REPORT THIS ISSUE).
```

Testing with version 2022.3.2-1, and the same package installs fine on macOS 12.2. `openssl` has been updated from LibreSSL 2.8.3 to LibreSSL 3.3.5.

**Comment by PicoMitchell** [comment]
Thank you for reporting this issue and for the details on the LibreSSL versions!

I may not get a chance to investigate this until need week, but I'll post here when I figure out what's going on with the `openssl` changes in 12.3.

**Comment by PicoMitchell** [comment]
After some quick searching of that error message, my first suspicion is a change in the default digest format with the new LibreSSL version (explained here when encrypting and decrypting data between different OpenSSL and LibreSSL versions: https://github.com/libressl-portable/portable/issues/378).

Since I don't have a macOS 12.3 beta installation handy at the moment, would you mind checking if the following command decrypts properly on macOS 12.3?

`echo 'U2FsdGVkX19eAyyWNQ6Vxh7AvlFAqAloJZjwhOvkjs4=' | openssl enc -d -aes-256-cbc -a -A -pass file:<(echo 'password')`

This was encrypted by manually specifying `-md sha256` during encryption on macOS 12.2.1, which I believe is the new default digest format, changed from using "md5". Here is the command used to encrypt this on macOS 12.2.1: `echo 'this is a test' | openssl enc -aes-256-cbc -a -A -md sha256 -pass file:<(echo 'password')`

If this works on macOS 12.3, I believe the solution that should hopefully work on all versions of macOS will be to specify `-md sha256` for all encryptions and decryptions, but I'll have to test that back to macOS 10.13 before I release an update.

**Comment by PicoMitchell** [comment]
Which beta version of macOS 12.3 were you testing on? I just had a chance to test on a clean installation of macOS 12.3 beta 5 (build 21E5227a) and was able to properly install a `mkuser` user creation package made on macOS 12.2.1 with mkuser 2022.3.2-1. I was also able to see that openssl/LibreSSL is still at version 2.8.3 in macOS 12.3 beta 5.

Regardless, I may still manually specify `-md sha256` (or maybe `-md sha512`) in all encryptions and decryptions to use a more secure digest than "md5" (which is used by openssl/LibreSSL 2.8.3) and to make things more future-proof. Even if Apple doesn't upgrade LibreSSL in macOS 12.3, they may upgrade it in macOS 13 or some other future version. I also got a chance to test manually specifying `-md sha256` (and `-md sha512`) back on macOS 10.13 and it works fine for backwards compatibility as well.

**Comment by MagerValp** [comment]
```
% sw_vers -buildVersion
21E5196i
% echo 'U2FsdGVkX19eAyyWNQ6Vxh7AvlFAqAloJZjwhOvkjs4=' | openssl enc -d -aes-256-cbc -a -A -pass file:<(echo 'password')
this is a test
```
👍🏼

I think it might be beta 1 on this tester laptop. I can try upgrading to a newer beta next week.

**Comment by PicoMitchell** [comment]
Thanks for this info! I wouldn't be surprised if Apple rolled back updating LibreSSL in macOS 12.3 because of things like this breaking without warning.

**Comment by PicoMitchell** [comment]
For reference, here are the release notes from LibreSSL 2.9.1 when the default message digest for `openssl enc` was changed from "md5" to "sha256": https://ftp.openbsd.org/pub/OpenBSD/LibreSSL/libressl-2.9.1-relnotes.txt

```
 * Compatibility Changes
   ...
   - Changed the default digest type of openssl(1) enc to sha256.
   ...
```

**Comment by timsutton** [comment]
Seems like it got rolled back in b5?


➜  /usr/bin/openssl version  
LibreSSL 2.8.3

➜  sw_vers
ProductName:	macOS
ProductVersion:	12.3
BuildVersion:	21E5227a

**Comment by PicoMitchell** [comment]
I decided to go ahead and specify `-md sha512` for all `openssl enc` encryption and decryption commands used for password obfuscation when creating and installing a user creation package (with the `--package` option) and just released this change in version **2022.3.7-1** which can be downloaded on the Releases page: https://github.com/freegeek-pdx/mkuser/releases

This means the `openssl enc` commands are now more future-proof and should always work regardless if Apple ever updates openssl/LibreSSL beyond version 2.8.3 in a future version of macOS (which would change the default message digest type from "md5" to "sha256"), and is also more secure than the current default of message digest type of "md5".
